src/Blocks/BlockHandler.h

```diff
@@ -123,7 +123,14 @@ class cBlockHandler
 	/** Returns if this block drops if it gets destroyed by an unsuitable situation.
 	Default: true */
 	virtual bool DoesDropOnUnsuitable(void);
-	
+
+	/** Returns the block's height */
+	virtual double GetBlockHeight(void);
+
+	/** Tests if a_Position is inside the block where a_Position is always [0..1[
+	Note that this is considered from a "top-down" perspective i.e. a top-half slab will always return true */
+	virtual bool IsInsideBlock(const Vector3d & a_Position, const NIBBLETYPE & a_BlockMeta);
```

Comment:
Also should be noted that `a_Position` is assumed to be within the block's grid boundaries.